### PR TITLE
initial connect4 game implementation

### PR DESCRIPTION
**Pull request by btbasham**

### CODE REVIEW

File: game/connect4.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    """
+    board: numpy array of shape NUM_ROWS, NUM_COLUMNS, where the value of
+    board[i, j] is: -1 if the hole in row i and column j is empty,
+                   0 if it is filled by the first player,
+                   1 if it is filled by the second player.
+
+    board[:, 0] corresponds to the BOTTOM of the board.
+
+    first_player_moves: list of columns in which the first player moved, in order.
```
Comment (amyzhang) [inline game/connect4.py:19]: This should be more clear. From the code it looks like "moves" is just a tuple of first_player_moves and second_player_moves, but is this list a tuple of coordinates?

Comment (btbasham) [inline game/connect4.py:19]: yeah, moves (only an argument to the constructor) is (first_player_moves, second_player_moves)

Comment (btbasham) [inline game/connect4.py:19]: I'll add a comment to the constructor

File: game/connect4.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        for row in range(NUM_ROWS):
+            if self.board[row, column] == -1:
+                return row
+        return -1
+
+    def make_move(self, player, column):
+        assert(player == self.current_turn())
+        assert(self.valid_moves()[column] == 1)
+        row = self.lowest_open_in_column(column)
+        self.board[row, column] = player
```
Comment (amyzhang) [inline game/connect4.py:79]: Need to check that lowest_open_in_column doesn't return -1. otherwise this will assign the last row in the board to player.

**Comment by amyzhang** [comment]
lgtm

**Pull request merged**